src/Feature/Redirects/code/Pipelines/RedirectResolver.cs

```diff
@@ -12,13 +12,13 @@ namespace Helixbase.Feature.Redirects.Pipelines
 {
     public class RedirectResolver : HttpRequestProcessor
     {
-        private readonly IContextRepository _contextRepository;
-        private readonly IContentRepository _contentRepository;
+        private readonly Func<IContextRepository> _contextRepositoryThunk;
+        private readonly Func<IContentRepository> _contentRepositoryThunk;
 
-        public RedirectResolver(IContextRepository contextRepository, IContentRepository contentRepository)
+        public RedirectResolver(Func<IContextRepository> contextRepositoryThunk, Func<IContentRepository> contentRepositoryThunk)
```

Comment:
Have you tested this locally? I wasn't aware that the default container supported auto-factory/Func injection

Reply:
Yes, we discovered the problem with the database being stored at app start up and found Corey's article when looking for a solution. Sitecore is using the .net core IServiceProvider container which is available in .net 4.7+